api/printful/printful.php

```diff
@@ -1,19 +1,18 @@
 <?php
-
+require_once(__DIR__ . '/../../template/top.php');
+require_once(__DIR__ . '/../api-cache.php');
 class PrintfulCustomAPI {
     private $api_key;
-
     public function __construct($printful_api_key = PRINTFUL_API_KEY) {
         $this->api_key = $printful_api_key;
     }
 
     protected function send_request($URI, $data = false) {
         $ch = curl_init();
-
         $headers = array();
         $headers[] = 'Authorization: Bearer ' . $this->api_key;
-
-        curl_setopt($ch, CURLOPT_URL, 'https://api.printful.com/' . $URI);
+        $endpoint = 'https://api.printful.com/' . $URI;
+        curl_setopt($ch, CURLOPT_URL, $endpoint);

```

Comment:
It seems tautological to pass `$endpoint` into the `get_valid_cache_entry` method and also set the endpoint before passing this in, we should probably just move this inside of the method.